actions/actions.go

```diff
@@ -0,0 +1,217 @@
+package actions
+
+import (
+	"bufio"
+	"fmt"
+	"github.com/SUSE/saptune/app"
+	"github.com/SUSE/saptune/sap/note"
+	"github.com/SUSE/saptune/system"
+	"io"
+	"os"
+	"strings"
+)
+
+//define constants and variables for the whole package
+const (
+	SaptuneService       = "saptune.service"
+	SapconfService       = "sapconf.service"
+	TunedService         = "tuned.service"
+	NoteTuningSheets     = "/usr/share/saptune/notes/"
+	OverrideTuningSheets = "/etc/saptune/override/"
+	ExtraTuningSheets    = "/etc/saptune/extra/" // ExtraTuningSheets is a directory located on file system for external parties to place their tuning option files.
+	setGreenText         = "\033[32m"
+	setRedText           = "\033[31m"
+	resetTextColor       = "\033[0m"
+	exitSaptuneStopped   = 1
+	exitNotTuned         = 3
+	footnote1X86         = "[1] setting is not supported by the system"
+	footnote1IBM         = "[1] setting is not relevant for the system"
+	footnote2            = "[2] setting is not available on the system"
+	footnote3            = "[3] value is only checked, but NOT set"
+	footnote4            = "[4] cpu idle state settings differ"
+	footnote5            = "[5] expected value does not contain a supported scheduler"
+	footnote6            = "[6] grub settings are mostly covered by other settings. See man page saptune-note(5) for details"
+	footnote7            = "[7] parameter value is untouched by default"
+)
+
+// RPMVersion is the package version from package build process
+var RPMVersion = "undef"
+
+// RPMDate is the date of package build
+var RPMDate = "undef"
+
+// set 'unsupported' footnote regarding the architecture
+var footnote1 = footnote1X86
+
+// Collection of tuning options from SAP notes and 3rd party vendors.
+var tuningOptions = note.GetTuningOptions(NoteTuningSheets, ExtraTuningSheets)
+
+// SelectAction selects the choosen action depending on the first command line
+// argument
+func SelectAction(stApp *app.App, saptuneVers string) {
+	switch system.CliArg(1) {
+	case "daemon":
+		DaemonAction(system.CliArg(2), saptuneVers, stApp)
+	case "note":
+		NoteAction(system.CliArg(2), system.CliArg(3), system.CliArg(4), stApp)
+	case "solution":
+		SolutionAction(system.CliArg(2), system.CliArg(3), stApp)
+	case "revert":
+		RevertAction(os.Stdout, system.CliArg(2), stApp)
+	case "service":
+		ServiceAction(system.CliArg(2), saptuneVers, stApp)
+	default:
+		PrintHelpAndExit(1)
+	}
+}
+
+// RevertAction Revert all notes and solutions
+func RevertAction(writer io.Writer, actionName string, tuneApp *app.App) {
+	if actionName != "all" {
+		PrintHelpAndExit(1)
+	}
+	fmt.Fprintf(writer, "Reverting all notes and solutions, this may take some time...\n")
+	if err := tuneApp.RevertAll(true); err != nil {
+		system.ErrorExit("Failed to revert notes: %v", err)
+		//panic(err)
```

Comment:
commented code here

Reply:
yes, you are right. Will be done with the next PR. Thanks